### PR TITLE
Implementing Padding support

### PR DESCRIPTION
**Pull request by infinytum**
I've added nice paddings for the demo dashboard too, makes it look a little better.

### CODE REVIEW

**Comment by infinytum** [comment]
Alright, we are testing the padding now

File: container/options.go [hunk 1]
```diff
@@ -403,3 +409,33 @@ func Bottom(opts ...Option) BottomOption {
 		return opts
 	})
 }
+
+type paddingOption func() []Option
```
Comment (mum4k) [inline container/options.go]: Do we need this type for anything? Looks like we should be good with the already existing definition of Option.

Comment (infinytum) [reply container/options.go]: Oops, thats useless, I was testing some other thing. Will remove.

File: container/options.go [hunk 2]
```diff
@@ -403,3 +409,33 @@ func Bottom(opts ...Option) BottomOption {
...
+
+func PaddingLeft(padd int) Option {
```
[the remark this answers is not in the document]
Comment (infinytum) [reply container/options.go]: Helper function is done, tests are in the works!

[the remark this answers is not in the document]
Comment (infinytum) [reply container/options.go]: Tests are now there too. area.Shrink is yet to do though.

File: container/container_test.go [hunk 3]
```diff
@@ -469,6 +469,34 @@ func TestNew(t *testing.T) {
...
 		},
+		{
+			desc:     "draw widget with padding",
+			termSize: image.Point{11, 11},
+			container: func(ft *faketerm.Terminal) (*Container, error) {
+				return New(
+					ft,
+					Border(draw.LineStyleLight),
+					PaddingBottom(1),
+					PaddingRight(1),
```
[the remark this answers is not in the document]
Comment (infinytum) [reply container/container_test.go]: Specific tests are now implemented

File: container/container_test.go [hunk 4]
```diff
@@ -469,6 +469,34 @@ func TestNew(t *testing.T) {
...
 			},
 		},
+		{
+			desc:     "draw widget with padding",
+			termSize: image.Point{11, 11},
+			container: func(ft *faketerm.Terminal) (*Container, error) {
+				return New(
+					ft,
+					Border(draw.LineStyleLight),
+					PaddingBottom(1),
```
[the remark this answers is not in the document]
Comment (infinytum) [reply container/container_test.go]: 4 seperated test cases have been implemented. Actually found out top and bottom were swapped!

[the remark this answers is not in the document]
Comment (infinytum) [reply container/container_test.go]: So I added a feature to allow draw errors too. Cause I need to know if the widget drawing failed. Negatest is implemented now.

File: container/container_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (infinytum) [reply container/container_test.go]: I dont quite know how I can add child containers in a test.

File: container/container_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (mum4k) [reply container/container_test.go]: Glad to hear our tests started to pay for themselves!

File: container/container_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (mum4k) [reply container/container_test.go]: You can take a look at the existing tests in container_test, e.g. this one creates child containers:

https://github.com/mum4k/termdash/blob/1e1dc15c47adf691d0f5a698a19d0662f2b5ab14/container/container_test.go#L107

Basically, we just pass the corresponding Split* options to the New() function. I can provide a more concrete example if need be, let me know.

**Comment by mum4k** [comment]
There have been large changes to the codebase since we started (moving of files) and it would be unfair to expect you to rebase and resolve the conflicts.

Therefore I have taken what you started and implemented it in #166 

Thank you again for getting this started!

**Comment by infinytum** [comment]
Thank you!

Sorry, I totally forgot about my open pull-requests on github. Im deep in my finals right now and might resume work on github after them :)

**Comment by mum4k** [comment]
That is no problem at all @infinytum, best of luck with the finals! I will be more than happy to resume working with you once you pass all your exams ;)

**Pull request closed without merging**